### PR TITLE
Rewrite companion window tests using RTL

### LINKED ISSUE #3651: Rewrite Enzyme tests with react-testing-library: Companion window tests
- [x] `__tests__/src/components/AnnotationSettings.test.js`
- [x] `__tests__/src/components/AttributionPanel.test.js`
- [x] `__tests__/src/components/CanvasAnnotations.test.js`
- [x] `__tests__/src/components/CanvasInfo.test.js`
- [x] `__tests__/src/components/CanvasLayers.test.js`
- [x] `__tests__/src/components/CollectionInfo.test.js`
- [x] `__tests__/src/components/LabelValueMetadata.test.js`
- [x] `__tests__/src/components/LayersPanel.test.js`
- [x] `__tests__/src/components/ManifestInfo.test.js`
- [x] `__tests__/src/components/ManifestRelatedLinks.test.js`
- [ ] ~`__tests__/src/components/SearchHit.test.js`~
- [x] `__tests__/src/components/SearchPanel.test.js`
- [x] `__tests__/src/components/SearchPanelControls.test.js`
- [x] `__tests__/src/components/SearchPanelNavigation.test.js`
- [ ] ~`__tests__/src/components/SearchResults.test.js`~
- [x] `__tests__/src/components/SidebarIndexItem.test.js`
- [x] `__tests__/src/components/SidebarIndexList.test.js`
- [ ] ~`__tests__/src/components/SidebarIndexTableOfContents.test.js`~
- [x] `__tests__/src/components/SidebarIndexThumbnail.test.js`
- [x] `__tests__/src/components/ThumbnailCanvasGrouping.test.js`
- [x] `__tests__/src/components/WindowSideBarAnnotationsPanel.test.js`
- [x] `__tests__/src/components/WindowSideBarButtons.test.js`
- [x] `__tests__/src/components/WindowSideBarCanvasPanel.test.js`
- [x] `__tests__/src/components/WindowSideBarInfoPanel.test.js`

Extracted from #3597

### PR DESCRIPTION
**Pull request by cbeer**
Fixes #3651 (remaining files and the 2 challenging tests here have been split to #3673)

### CODE REVIEW

File: __tests__/utils/store.js [hunk 1]
```diff
@@ -11,7 +14,7 @@ export function renderWithProviders(
   {
     preloadedState = {},
     // Automatically create a store instance if no store was passed in
-    store = createPluggableStore(preloadedState, []),
+    store = createStore(rootReducer, preloadedState),
```
Comment (marlo-longley) [inline __tests__/utils/store.js:17]: Can/should you rebase this which is already in master?

File: __tests__/src/components/CanvasAnnotations.test.js [hunk 2]
```diff
@@ -41,28 +40,30 @@ describe('CanvasAnnotations', () => {
...
+    createWrapper({ annotations });
 
-    expect(wrapper.find(MenuList).length).toEqual(1);
-    expect(wrapper.find(MenuItem).length).toEqual(2);
+    expect(screen.getByRole('menu')).toBeInTheDocument();
+    expect(screen.getAllByRole('menuitem').length).toEqual(2);
   });
 
-  it('scrolls to the selected annotation', () => {
+  xit('scrolls to the selected annotation', () => {
```
Comment (marlo-longley) [inline __tests__/src/components/CanvasAnnotations.test.js:66]: just pointing this `xit` out -- maybe was difficult with ScrollTo?

File: __tests__/src/components/CanvasLayers.test.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-import Typography from '@material-ui/core/Typography';
-import { DragDropContext, Droppable, Draggable } from '@hello-pangea/dnd';
+import userEvent from '@testing-library/user-event';
+import { Resource } from 'manifesto.js';
+import { DragDropContext } from '@hello-pangea/dnd';
+import { renderWithProviders } from '../../utils/store';
 import { CanvasLayers } from '../../../src/components/CanvasLayers';
-import IIIFThumbnail from '../../../src/containers/IIIFThumbnail';
 
 /** Utility function to wrap CanvasAnnotations */
```
Comment (marlo-longley) [inline __tests__/src/components/CanvasLayers.test.js:9]: This is CanvasLayers not CanvasAnnotations -- minor but could confuse ?

File: __tests__/src/components/SearchHit.test.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ describe('SearchHit', () => {
     expect(wrapper.find('WithStyles(ForwardRef(ListItem))').length).toEqual(1);
     expect(wrapper.find('WithStyles(ForwardRef(ListItem))').prop('selected')).toEqual(true);
     expect(wrapper.find('WithStyles(ForwardRef(ListItem))').prop('className')).toEqual('windowSelected');
-    expect(wrapper.find('WithStyles(ForwardRef(ListItemText))').render().text()).toEqual('1Light up the moose , and start the chai ');
+    expect(wrapper.find('WithStyles(ForwardRef(ListItemText))').render().text()).toEqual('1Light up the moose , and start the chai more');
```
Comment (marlo-longley) [inline __tests__/src/components/SearchHit.test.js:37]: !

File: __tests__/src/components/SearchPanelControls.test.js [hunk 5]
```diff
@@ -1,16 +1,13 @@
...
-import Input from '@material-ui/core/Input';
-import TextField from '@material-ui/core/TextField';
-import SearchIcon from '@material-ui/icons/SearchSharp';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { renderWithProviders } from '../../utils/store';
 import { SearchPanelControls } from '../../../src/components/SearchPanelControls';
 
 /**
  * Helper function to create a shallow wrapper around AttributionPanel
```
Comment (marlo-longley) [inline __tests__/src/components/SearchPanelControls.test.js:7]: Another minor copy issue in comment

File: __tests__/src/components/SearchPanelNavigation.test.js [hunk 6]
```diff
@@ -17,26 +18,28 @@ function createWrapper(props) {
...
-      expect(wrapper.find('WithStyles(ForwardRef(Typography))').text()).toEqual('pagination');
-      expect(wrapper.find('WithWorkspaceContext(WithPlugins(MiradorMenuButton))[disabled=false]').length).toEqual(2);
-      wrapper.find('WithWorkspaceContext(WithPlugins(MiradorMenuButton))[disabled=false]').first().props().onClick();
+      expect(screen.getByText('pagination')).toBeInTheDocument();
+      expect(screen.getAllByRole('button').length).toEqual(2);
+      await user.click(screen.getByRole('button', { name: 'searchPreviousResult' }));
       expect(selectAnnotation).toHaveBeenCalledWith('1');
-      wrapper.find('WithWorkspaceContext(WithPlugins(MiradorMenuButton))[disabled=false]').last().props().onClick();
+      await user.click(screen.getByRole('button', { name: 'searchNextResult' }));
       expect(selectAnnotation).toHaveBeenCalledWith('3');
```
Comment (marlo-longley) [inline __tests__/src/components/SearchPanelNavigation.test.js:34]: Sorry if I'm missing something but -- selected is 2. We click searchPrevious and get 1. Then we hit searchNext and get 3? Shouldn't we be back on 2?

File: src/components/WindowSideBarCanvasPanel.js [hunk 7]
```diff
@@ -115,6 +115,7 @@ export class WindowSideBarCanvasPanel extends Component {
                     name="sequenceId"
                     classes={{ select: classes.select }}
                     className={classes.selectEmpty}
+                    data-testid="sequence-select"
```
Comment (marlo-longley) [inline src/components/WindowSideBarCanvasPanel.js:118]: @phil-plencner-hl this is probably how you will need to access that inner-rendered component we were looking at -- add the data-testid directly in the component.

**Review by marlo-longley — APPROVED** [review]
Thanks for all this work !! 
I can approve -- some super minor text changes and 1 question.

File: __tests__/src/components/CanvasAnnotations.test.js (hunk 2 above)
Comment (marlo-longley) [reply __tests__/src/components/CanvasAnnotations.test.js:66]: Update, I found a few more `xits` and think they are fine for now.

**Comment by cbeer** [comment]
I think it'd be good to address the pre-existing comment problems in a separate PR.

File: __tests__/src/components/SearchPanelNavigation.test.js (hunk 6 above)
Comment (cbeer) [reply __tests__/src/components/SearchPanelNavigation.test.js:34]: Except we're not actually updating the underlying state, so we never go anywhere 🤷‍♂️

**Comment by marlo-longley** [comment]
Yes! It's super minor.

**Pull request merged**